hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -219,4 +219,11 @@ public Map<String, Object> updateSchemaAndResetOrderingValInMetadata(Map<String,
   public long extractRecordPosition(T record, Schema schema, String fieldName, long providedPositionIfNeeded) {
     return providedPositionIfNeeded;
   }
+
+  /**
+   * Constructs engine specific delete record.
+   */
+  public T constructRawDeleteRecord(Map<String, Object> metadata) {
```

Comment:
@danny0405, where do you suggest to implement this function for Flink related code?

Reply:
what is a raw delete record? A empty record with record key?

Reply:
Yeah, It is supposed to be like that. Each query engine should implement their own delete record.

Reply:
A payload from the `DeleteBlock` contains only record keys, so the engine needs to decode it as an native record with record key set up. Any other deletes records from the data block should includes all the data fields, we should keep these fields while deciphering.